packages/flutter_markdown/lib/src/builder.dart

```diff
@@ -85,6 +85,84 @@ abstract class MarkdownBuilderDelegate {
   TextSpan formatText(MarkdownStyleSheet styleSheet, String code);
 }
 
+/// An anchor in markdown is basically a marker inside the markdown text
+/// with an id.
+/// We can use this to scroll to a specific place inside the text.
+///
+/// When using [md.ExtensionSet.gitHubWeb] (more specifically
+/// [md.HeaderWithIdSyntax] and [md.SetextHeaderWithIdSyntax]) anchors are
+/// automatically created for headings.
+///
+/// We can then use the [AnchorController] to scroll to a specific header and
+/// see which headers are currently visible.
+///
+/// Usually one can also create anchors in markdown via the html tag `a` but as
+/// we can't render HTML inside flutter currently (as far as I know) only
+/// headers should have an anchor.
+class AnchorData {
+  AnchorData(this.id, this.text);
+
+  /// The id of the anchor e.g. 'table-of-contents'.
+  final String id;
+
+  /// The name of the anchor (so the text of the heading if the anchor is for a
+  /// heading).
+  /// E.g. 'Table of contents'.
+  final String text;
+
+  @override
+  String toString() => 'AnchorData(id: $id, text: $text)';
+}
+
+/// [AnchorData] with an index.
+///
+/// The index is the position inside `children` passed by
+/// [MarkdownWidget.build].
+/// We need it to scroll to a specific anchor via [ScrollablePositionedList].
+class IndexedAnchorData extends AnchorData {
```

Comment:
I'm curious why you implemented this as a IS-A relationship instead of a HAS-A relationship?

Reply:
Ah, don't read too much into it, I just tried different stuff when prototyping and used the first thing that more or less worked (altough it looks polished because of the comments😁). I'll change it if you want:)

Reply:
I haven't studied your implementation, but I'm assuming that the `AnchorData` captures configuration data from the user of the library, while the `IndexedAnchorData` is derived from the user supplied `AnchorData` and the content stream? So it'd make sense to me for the `IndexedAnchorData` to refer to the supplied `AchorData`, not copy it's data.

Reply:
No, the user only inputs the markdown text, the rest is all derived from that. I create the `AnchorData` from the AST from the `markdown` package.

Reply:
Okiday, sgtm